test/lib/validation/schema/__helper__/builder-bundleOptions.js

```diff
@@ -13,7 +13,128 @@ export default {
 	 */
 	defineTests: function(test, assertValidation, type) {
 		// Version specific tests
-		SpecificationVersion.getVersionsForRange(">=3.0").forEach(function(specVersion) {
+		SpecificationVersion.getVersionsForRange(">=4.0").forEach(function(specVersion) {
+			test(`${type} (specVersion ${specVersion}): builder/bundles/bundleOptions`, async (t) => {
+				await assertValidation(t, {
+					"specVersion": specVersion,
+					"kind": "project",
+					"type": type,
+					"metadata": {
+						"name": "com.sap.ui5.test",
+						"copyright": "yes"
+					},
+					"builder": {
+						"bundles": [{
+							"bundleOptions": {
+								"optimize": false,
+								"decorateBootstrapModule": false,
+								"addTryCatchRestartWrapper": true,
+								"numberOfParts": 8,
+								"sourceMap": false
+							}
+						}]
+					}
+				});
+			});
+
+			test(`${type} (specVersion ${specVersion}): builder/bundles/bundleOptions properties removal`,
+				async (t) => {
+					await assertValidation(t, {
+						"specVersion": specVersion,
+						"kind": "project",
+						"type": type,
+						"metadata": {
+							"name": "com.sap.ui5.test",
+							"copyright": "yes"
+						},
+						"builder": {
+							"bundles": [{
+								"bundleOptions": {
+									"debugMode": true
```

Comment:
```suggestion
									"usePredefineCalls": true
```

debugMode removal is already tested further down